packages/expo-modules-core/ios/Core/DynamicTypes/DynamicEnumType.swift

```diff
@@ -18,6 +18,9 @@ internal struct DynamicEnumType: AnyDynamicType {
   }
 
   func cast<ValueType>(_ value: ValueType, appContext: AppContext) throws -> Any {
+    if let value = value as? any Enumerable {
+      return value
+    }
```

Comment:
I think it's not entirely correct. The value could be an enumerable but of a different type that the `DynamicEnumType` was initialized with.

Reply:
After chatting, let's merge for now, but it's not an ideal fix.